src/entities/users/user.schema.ts

```diff
@@ -1,18 +1,54 @@
-import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Prop, Schema, SchemaFactory, raw } from '@nestjs/mongoose';
 import { HydratedDocument } from 'mongoose';
+import { Access, Modules, } from './types/userTypes';
 
-export type UserDocument = HydratedDocument<User>;
+const moduleType = [
```

Comment:
It should be refactored. If we add a new access type or remove one (in AccessType), we don't need to change moduleType. Can we improve it?